packages/run-protocol/src/reserve/collateralReserve.js

```diff
@@ -198,9 +205,13 @@ const start = async (zcf, privateArgs) => {
   );
   /** @typedef {typeof publicFacet} ReservePublicFacet */
 
+  // @ts-expect-error cast
   return harden({ creatorFacet, publicFacet });
 };
 
 harden(start);
 
 export { start };
+
+/** @typedef {Required<ReturnType<typeof start>>['publicFacet']} CollateralReservePublicFacet */
+/** @typedef {Required<ReturnType<typeof start>>['creatorFacet']} CollateralReserveCreatorFacet */
```

Comment:
what does `Required<>` do here?

Reply:
Make every optional member no longer be optional.